lib/dao.js

```diff
@@ -2689,7 +2689,23 @@ DataAccessObject.updateAll = function(where, data, options, cb) {
     Model.notifyObserversOf('before save', context,
       function(err, ctx) {
         if (err) return cb(err);
-        doUpdate(ctx.where, ctx.data);
+
+        data = ctx.data;
+        var update = data;
```

Comment:
Try with this:
```js
        data = ctx.data;
        var inst = data;
        if (!(data instanceof Model)) {
          inst = new Model(data, {applyDefaultValues: false});
        }

        // validation required
        inst.isValid(function(valid) {
          if (valid) {
            doUpdate(ctx.where, ctx.data);
          } else {
            cb(new ValidationError(inst), inst);
          }
        }, options);
      });
```
You will see that it is not affected and works fine, because `inst` variable is assigned to be equal to `data` as well .. so it is either no needed at all, or the test coverage is not covering this part .. I think it is not needed, because `inst` variable is doing the job.